File: wagerproof-ios-native/Wagerproof/Features/Analytics/Components/BullpenFatigueCard.swift
import SwiftUI
import WagerproofDesign
import WagerproofModels

/// One bullpen in the fatigue/trend list. OVERWORKED (red) vs DECLINING
/// (amber) follows the row's `flag`. IP thresholds (13 over 3 days, 22 over
/// 5 days) match the RN highlight rules.
struct BullpenFatigueCard: View {
    let bullpen: MLBBullpenFatigue

    var body: some View {
        let overworked = bullpen.flag == "overworked"
        let color = overworked ? Regression.lossRed : Regression.warnAmber
        RegressionAccentRow(color: color) {
            VStack(alignment: .leading, spacing: 10) {
                HStack {
                    Text(bullpen.teamName)
                        .font(.system(size: 14, weight: .bold))
                        .foregroundStyle(Color.appTextPrimary)
                    Spacer()
                    RegressionPill(text: overworked ? "OVERWORKED" : "DECLINING", color: color)
                }
                HStack(spacing: 10) {
                    RegressionStat(
                        label: "IP L3d",
                        value: String(format: "%.1f", bullpen.bpIpLast3d),
                        color: bullpen.bpIpLast3d >= 13 ? Regression.lossRed : nil
                    )
                    RegressionStat(
                        label: "IP L5d",
                        value: String(format: "%.1f", bullpen.bpIpLast5d),
                        color: bullpen.bpIpLast5d >= 22 ? Regression.lossRed : nil
                    )
                    RegressionStat(
                        label: "SEASON xFIP",
                        value: bullpen.seasonBpXfip.map { String(format: "%.2f", $0) } ?? "-"
                    )
                    RegressionStat(
                        label: "TREND xFIP",
                        value: bullpen.trendBpXfip.map { Regression.signed($0, decimals: 2) } ?? "-",
                        color: bullpen.trendBpXfip.map { $0 > 0 ? Regression.lossRed : Regression.winGreen }
                    )
                }
            }
        }
    }
}

File: wagerproof-android-native/app/src/main/java/com/wagerproof/app/features/analytics/BullpenFatigueCard.kt
package com.wagerproof.app.features.analytics

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.wagerproof.core.design.tokens.AppColors
import com.wagerproof.core.models.MLBBullpenFatigue
import java.util.Locale

/**
 * One bullpen in the fatigue/trend list. OVERWORKED (red) vs DECLINING (amber)
 * follows the row's `flag`. IP thresholds (13 over 3d, 22 over 5d) highlighted
 * red. Port of iOS `BullpenFatigueCard`.
 */
@Composable
fun BullpenFatigueCard(bullpen: MLBBullpenFatigue, modifier: Modifier = Modifier) {
    val overworked = bullpen.flag == "overworked"
    val color = if (overworked) Regression.lossRed else Regression.warnAmber
    RegressionAccentRow(color = color, modifier = modifier) {
        Column(Modifier.fillMaxWidth(), verticalArrangement = Arrangement.spacedBy(10.dp)) {
            Row(verticalAlignment = Alignment.CenterVertically) {
                Text(bullpen.teamName, fontSize = 14.sp, fontWeight = FontWeight.Bold, color = AppColors.appTextPrimary, modifier = Modifier.weight(1f))
                RegressionPill(if (overworked) "OVERWORKED" else "DECLINING", color)
            }
            Row(horizontalArrangement = Arrangement.spacedBy(10.dp)) {
                RegressionStat(
                    "IP L3d",
                    String.format(Locale.US, "%.1f", bullpen.bpIpLast3d),
                    Modifier.weight(1f),
                    if (bullpen.bpIpLast3d >= 13) Regression.lossRed else null,
                )
                RegressionStat(
                    "IP L5d",
                    String.format(Locale.US, "%.1f", bullpen.bpIpLast5d),
                    Modifier.weight(1f),
                    if (bullpen.bpIpLast5d >= 22) Regression.lossRed else null,
                )
                RegressionStat("SEASON xFIP", bullpen.seasonBpXfip?.let { String.format(Locale.US, "%.2f", it) } ?: "-", Modifier.weight(1f))
                RegressionStat(
                    "TREND xFIP",
                    bullpen.trendBpXfip?.let { Regression.signed(it, 2) } ?: "-",
                    Modifier.weight(1f),
                    bullpen.trendBpXfip?.let { if (it > 0) Regression.lossRed else Regression.winGreen },
                )
            }
        }
    }
}
